cpp/daal/src/algorithms/dtrees/forest/df_training_parameter.cpp

```diff
@@ -66,6 +66,7 @@ Status checkImpl(const decision_forest::training::interface2::Parameter & prm)
     DAAL_CHECK_EX((prm.minWeightFractionInLeafNode >= 0.0) && (prm.minWeightFractionInLeafNode <= 0.5), ErrorIncorrectParameter, ParameterName,
                   minWeightFractionInLeafNodeStr());
     DAAL_CHECK_EX(prm.minImpurityDecreaseInSplitNode >= 0.0, ErrorIncorrectParameter, ParameterName, minImpurityDecreaseInSplitNodeStr());
+    DAAL_CHECK_EX((prm.maxLeafNodes == 0) || (prm.maxLeafNodes > 1), ErrorIncorrectParameter, ParameterName, maxLeafNodesStr());
```

Comment:
Why maxLeafNodes == 1 is not valid?

Reply:
Here check is similar to Sklearn, we may finalize it after offline discussion. The idea is: an user set maxLeafNodes, in implementation remainingSplitNodes is used (remainingSplitNodes = maxLeafNodes - 1, in case maxLeafNodes>=2) and after each split remainingSplitNodes decreased to zero, other nodes are leaf

Reply:
Discussed offline. We can easily support maxLeafNodes == 1 case. It also simplifies the logic and our checks

Reply:
it was done